test/test_prototype_transforms.py

```diff
@@ -222,6 +231,58 @@ def test_random_resized_crop(self, transform, input):
         transform(input)
 
 
+@pytest.mark.parametrize(
+    "sample",
+    [
+        [make_pil_image(), make_vanilla_tensor_image(), make_vanilla_tensor_image()],
+        [make_vanilla_tensor_image(), make_pil_image(), make_vanilla_tensor_image()],
+        [make_vanilla_tensor_image(), make_vanilla_tensor_image(), make_pil_image()],
+        [make_image(), make_vanilla_tensor_image(), make_vanilla_tensor_image()],
+        [make_vanilla_tensor_image(), make_image(), make_vanilla_tensor_image()],
+        [make_vanilla_tensor_image(), make_vanilla_tensor_image(), make_image()],
+    ],
+)
+def test_simple_tensor_heuristic(sample):
+    def split_on_simple_tensor(to_split):
+        simple_tensors = []
+        others = []
+        for item, predicate in zip(to_split, sample):
```

Comment:
I'm not sure I understand the naming scheme "item" and "predicate" here?

Reply:
Yeah, I need to find better names ...

Reply:
I've added more comments and renamed the `predicate`. LMK if it is now clearer.